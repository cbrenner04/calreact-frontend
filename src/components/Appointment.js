import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import PropTypes from 'prop-types';
import $ from 'jquery';

import {formatDate} from '../utils/format';

export default class Appointment extends Component {
  constructor (props) {
    super(props)
    this.state = {
        appointment: props.appointment
    }
  }

  static propTypes = {
    appointment: PropTypes.object.isRequired
  }

  static defaultProps = {
    appointment: {}
  }

  componentDidMount() {
    if (this.props.match && sessionStorage.getItem('user')) {
      $.ajax({
        type: "GET",
        url: `http://localhost:3000/appointments/${this.props.match.params.id}`,
        dataType: "JSON",
        headers: JSON.parse(sessionStorage.getItem('user')),
      }).done((data) => {
        this.setState({appointment: data})
      })
    }
  }

  render() {
    return (
        <div className='appointment'>
          <Link to={ `/appointments/${this.state.appointment.id}` }>
              <h3>{ this.state.appointment.title }</h3>
          </Link>
          <p>{ formatDate(this.state.appointment.appt_time) }</p>
          <Link to={ `/appointments/${this.state.appointment.id}/edit` }>
            Edit
          </Link>
        </div>
    )
  }
}
