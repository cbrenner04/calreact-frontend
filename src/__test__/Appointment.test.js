import React from 'react';
import {BrowserRouter as Router} from 'react-router-dom';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';

import Appointment from '../components/Appointment';

it('renders without crashing', () => {
  shallow(<Appointment />);
});

describe('render', () => {
  it('should display the appointment title', () => {
    const appointment = mount(
      <Router>
        <Appointment
          appointment={{ id: 1, title: 'Team standup', appt_time: new Date() }}
        />
      </Router>
    );
    const title = <h3>Team standup</h3>;
    expect(appointment.contains(title)).toEqual(true);
  });

  it('should display the appointment time', () => {
    const appointment = mount(
      <Router>
        <Appointment
          appointment={{
            id: 1,
            title: 'Team standup',
            appt_time: new Date('04/11/2017, 12:00:00')
          }}
        />
      </Router>
    );
    const appt_time = <p>April 11 2017, 12:00:00 pm</p>;
    expect(appointment.contains(appt_time)).toEqual(true);
  });

  it('renders Appointment correctly', () => {
    const appointment = renderer.create(
      <Router>
        <Appointment
          appointment={{
            id: 1,
            title: 'Team standup',
            appt_time: new Date('04/11/2017, 12:00:00')
          }}
        />
      </Router>
    );
    let tree = appointment.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

