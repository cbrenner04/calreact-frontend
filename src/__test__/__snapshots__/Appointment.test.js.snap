// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders Appointment correctly 1`] = `
<div
  className="appointment"
>
  <a
    href="/appointments/1"
    onClick={[Function]}
  >
    <h3>
      Team standup
    </h3>
  </a>
  <p>
    April 11 2017, 12:00:00 pm
  </p>
  <a
    href="/appointments/1/edit"
    onClick={[Function]}
  >
    Edit
  </a>
</div>
`;
